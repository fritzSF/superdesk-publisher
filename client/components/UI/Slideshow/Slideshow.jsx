import React from "react";
import PropTypes from "prop-types";
import AwesomeSlider from "react-awesome-slider";
import AwsSliderStyles from "./slideshow.css";

const Slideshow = ({ tenant, items, type = "viewImage" }) => {
  let base = tenant.subdomain
    ? "https://" + tenant.subdomain + "." + tenant.domain_name
    : "https://" + tenant.domain_name;

  let images = [];

  items.forEach(item => {
    if (item.article_media && item.article_media.image) {
      let rendition = item.article_media.renditions.find(
        el => el.name === type
      );

      if (!rendition)
        rendition = item.article_media.renditions.find(
          el => el.name === "original"
        );

      let extension =
        rendition.image.file_extension === "jpeg"
          ? "jpg"
          : rendition.image.file_extension;
      images.push(
        base + "/media/" + rendition.image.asset_id + "." + extension
      );
    }
  });

  if (!images.length) return null;

  return (
    <AwesomeSlider cssModule={AwsSliderStyles}>
      {images.map((image, index) => (
        <div data-src={image} key={"slideshowItem" + index} />
      ))}
    </AwesomeSlider>
  );
};

Slideshow.propTypes = {
  tenant: PropTypes.object.isRequired,
  items: PropTypes.array.isRequired,
  type: PropTypes.string
};

export default Slideshow;
