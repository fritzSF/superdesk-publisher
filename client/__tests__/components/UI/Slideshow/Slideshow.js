import React from "react";
import Slideshow from "../../../../components/UI/Slideshow/Slideshow.jsx";
import { render } from "@testing-library/react";

const tenant = { subdomain: "test", domain_name: "sourcefabric.org" };

const items = [
  {
    article_media: {
      id: 2634,
      image: {
        id: 2246
      },
      renditions: [
        {
          width: 1322,
          height: 955,
          name: "original",
          id: 24967,
          image: {
            id: 2246,
            file_extension: "jpeg",
            asset_id:
              "2019020713028_aac0d4a7f9a46fc7f22d7e6cb0487092b6db9b7cd3091029d01e5a1be58bde45"
          }
        }
      ]
    }
  },
  {
    article_media: {
      id: 2635,
      image: {
        id: 2280
      },
      renditions: [
        {
          width: 1920,
          height: 969,
          name: "original",
          id: 24968,
          image: {
            id: 2280,
            file_extension: "png",
            asset_id:
              "2019030620038_3e78ef460274a0e076ee7800f8f4463053a76dc3e38f645808c0bd69a2895f1e"
          }
        }
      ]
    }
  }
];

describe("UI/Slideshow", () => {
  it("renders correcty", () => {
    const { container } = render(<Slideshow tenant={tenant} items={items} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
