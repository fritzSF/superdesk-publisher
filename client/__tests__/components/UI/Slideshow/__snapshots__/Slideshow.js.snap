// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Slideshow renders correcty 1`] = `
<div
  class="aws-sld aws-sld--organic-arrows"
>
  <div
    class="aws-sld__wrapper"
  >
    <div
      class="aws-sld__container"
    >
      <div
        class="aws-sld__box aws-sld--active"
      >
        <div
          class="aws-sld__content"
        >
          <img
            src="https://test.sourcefabric.org/media/2019020713028_aac0d4a7f9a46fc7f22d7e6cb0487092b6db9b7cd3091029d01e5a1be58bde45.jpg"
          />
        </div>
      </div>
      <div
        class="aws-sld__box"
      />
    </div>
    <div
      class="aws-sld__controls"
    >
      <button
        class="aws-sld__next"
      >
        <span
          class="aws-sld__controls__arrow-right"
        />
      </button>
      <button
        class="aws-sld__prev"
      >
        <span
          class="aws-sld__controls__arrow-left"
        />
      </button>
    </div>
  </div>
  <nav
    class="aws-sld__bullets"
  >
    <button
      class="aws-sld__bullets--active"
      data-index="0"
    >
      0
    </button>
    <button
      data-index="1"
    >
      1
    </button>
  </nav>
</div>
`;
